src/ussd_registration.js

```diff
@@ -0,0 +1,343 @@
+                ],
+                next: function(choice) {
+                    return go.utils
+                        .track_redials(self.contact, self.im, choice.value)
+                        .then(function() {
+                            if (choice.value === 'continue') {
+                                return {
+                                    name: creator_opts.name,
+                                    creator_opts: creator_opts
+                                };
+                                // return creator_opts.name;
+                            } else if (choice.value === 'restart') {
+                                return 'state_start';
+                            }
+                        });
+                }
+            });
+        });
+
+
+    // START STATE
+
+        self.add('state_start', function(name) {
+            return go.utils
+                .check_msisdn_hcp(self.im.user.addr)
+                .then(function(hcp_recognised) {
+                    if (hcp_recognised) {
+                        return self.states.create('state_msisdn');
+                    } else {
+                        return self.states.create('state_auth_code');
+                    }
+                });
+        });
+
+
+    // REGISTRATION STATES
+
+        // FreeText st-1
+        self.add('state_auth_code', function(name) {
+            return new FreeText(name, {
+                question: $(questions[name]),
+                check: function(content) {
+                    return go.utils
+                        .validate_personnel_code(self.im, content)
+                        .then(function(valid_personnel_code) {
+                            if (valid_personnel_code) {
+                                return null;  // vumi expects null or undefined if check passes
+                            } else {
+                                return $(get_error_text(name));
+                            }
+                        });
+                },
+                next: 'state_msisdn'
+            });
+        });
+
+        // FreeState st-02
+        self.add('state_msisdn', function(name) {
+            return new FreeText(name, {
+                question: $(questions[name]),
+                check: function(content) {
+                    if (go.utils.is_valid_msisdn(content)) {
+                        return null;  // vumi expects null or undefined if check passes
+                    } else {
+                        return $(get_error_text(name));
+                    }
+                },
+                next: 'state_msg_receiver'
+            });
+        });
+
+        // ChoiceState st-03
+        self.add('state_msg_receiver', function(name) {
+            return new ChoiceState(name, {
+                question: $(questions[name]),
+                choices: [
+                    new Choice('the_mother', $("The Mother")),
+                    new Choice('the_father', $("The Father")),
```

Comment:
Bike shedding - please just make it 'mother' and 'father', not 'the_mother' and 'the_father'